agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/SecondEntryPoint.java

```diff
@@ -197,46 +197,67 @@ public void customize(AutoConfigurationCustomizer autoConfiguration) {
 
     RpConfiguration rpConfiguration = FirstEntryPoint.getRpConfiguration();
     if (rpConfiguration != null) {
-      RpConfigurationPolling.startPolling(rpConfiguration, config, telemetryClient, appIdSupplier);
+      RpConfigurationPolling.startPolling(
+          rpConfiguration, configuration, telemetryClient, appIdSupplier);
     }
 
     // initialize StatsbeatModule
-    statsbeatModule.start(telemetryClient, config);
+    statsbeatModule.start(telemetryClient, configuration);
 
     AfterAgentListener.setAppIdSupplier(appIdSupplier);
 
     // TODO (trask) add this method to AutoConfigurationCustomizer upstream?
     ((AutoConfiguredOpenTelemetrySdkBuilder) autoConfiguration).registerShutdownHook(false);
 
     QuickPulse quickPulse;
-    if (config.preview.liveMetrics.enabled) {
+    if (configuration.preview.liveMetrics.enabled) {
       quickPulse =
           QuickPulse.create(
-              LazyHttpClient.newHttpPipeLineWithDefaultRedirect(config.preview.authentication),
+              LazyHttpClient.newHttpPipeLineWithDefaultRedirect(
+                  configuration.preview.authentication),
               () -> {
                 ConnectionString connectionString = telemetryClient.getConnectionString();
                 return connectionString == null ? null : connectionString.getLiveEndpoint();
               },
               telemetryClient::getInstrumentationKey,
               telemetryClient.getRoleName(),
               telemetryClient.getRoleInstance(),
-              config.preview.useNormalizedValueForNonNormalizedCpuPercentage,
+              configuration.preview.useNormalizedValueForNonNormalizedCpuPercentage,
               FirstEntryPoint.getAgentVersion());
     } else {
       quickPulse = null;
     }
     telemetryClient.setQuickPulse(quickPulse);
 
     autoConfiguration
+        .addPropertiesCustomizer(new AiConfigCustomizer())
+        .addSpanExporterCustomizer(
+            (spanExporter, config) -> {
+              if ("none".equals(config.getString("otel.traces.exporter"))) {
+                // in this case the spanExporter here is the noop spanExporter
+                return spanExporter;
+              } else {
+                return wrapSpanExporter(spanExporter, configuration);
```

Comment:
and other places (when they are both present) (?)
```suggestion
                return wrapSpanExporter(spanExporter, aiConfig);
```